docs/content/terraform.md

```diff
@@ -666,6 +666,248 @@ curl localhost:8181/v0/data/terraform/analysis/authz -d @tfplan.json
 curl localhost:8181/v0/data/terraform/analysis/authz -d @tfplan_large.json
 ```
 
+### 7. Create and save terraform module plan 
+
+Create a new Terraform file that includes a
+security group and security group from a module.  (This example uses the module from https://github.com/terraform-aws-modules/terraform-aws-security-group)
+
+```shell
+cat >main.tf <<EOF
+provider "aws" {
+  region = "us-east-1"
+}
+
+data "aws_vpc" "default" {
+  default = true
+}
+
+module "http_sg" {
+  source = "git::https://github.com/terraform-aws-modules/terraform-aws-security-group.git?ref=v3.10.0"
+
+  name        = "http-sg"
+  description = "Security group with HTTP ports open for everybody (IPv4 CIDR), egress ports are all world open"
+  vpc_id      = data.aws_vpc.default.id
+
+  ingress_cidr_blocks = ["0.0.0.0/0"]
+}
+
+
+resource "aws_security_group" "allow_tls" {
+  name        = "allow_tls"
+  description = "Allow TLS inbound traffic"
+  vpc_id      = data.aws_vpc.default.id
+
+  ingress {
+    description = "TLS from VPC"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["10.0.0.0/8"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "allow_tls"
+  }
+}
+EOF
+```
+
+Then initialize Terraform and ask it to calculate what changes it will make and store the output in `tfplan.binary`.
+
+```shell
+terraform init
+terraform plan --out tfplan.binary
+```
+
+### 8. Convert the new Terraform plan into JSON
+
+Repeat the commands from Step 2 to produce the json representation of the terraform plan
+
+```shell
+terraform show -json tfplan.binary > tfplan2.json
+```
+
+### 9. Write the OPA policy to collect resources
+
+The policy creates a function to find resources defined in the plan
+
+* Resources can be specified under the root module or in child modules
+* You may want evaluate against the combined group of these resources
+* This example is scoped to the planned changes section of the json representation
+
+The policy uses the walk keyword to explore the json structure, and uses conditions to filter for the specific paths where resources would be found
+
+**terraform_module.rego**:
+
+```shell
+package terraform.module
+
+resources[r] {
```

Comment:
nit: For performance reasons (until #822 is implemented) we might want to switch this to use a comprehension for resources like:

```rego
resources := { r |
...
} 
```

If it gets referenced in a lot of places it can make a pretty substantial difference.

Reply:
corrected